### PR TITLE
Bump go to 1.21 and pipeline to v0.59.0 LTS

### PR DESCRIPTION
**Pull request by khrm**
# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Update Go version to 1.21 and Pipelines to v0.59.0. Minimum Kubernetes version is 1.27.
```

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/kind misc

**Comment by khrm** [comment]
It seems I need to update test go version.

**Comment by khrm** [comment]
Should be fixed by https://github.com/tektoncd/plumbing/pull/1938

**Review by khrm — COMMENTED** [review]
/retest

**Comment by khrm** [comment]
We also need to change replace in go.mod file.

**Comment by khrm** [comment]
"error: RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly: CANCEL (err 8)
" Not sure why we are getting this.

**Comment by khrm** [comment]
@savitaashture Can you review this?

File: go.mod [hunk 1]
```diff
@@ -1,57 +1,57 @@
 module github.com/tektoncd/triggers
 
-go 1.19
+go 1.21
```
Comment (savitaashture) [inline go.mod:3]: should we add/update a doc with golang min version requirement and k8s min version requirement ?

Comment (khrm) [reply go.mod:3]: We don't mention Golang min version. We just give build image which can run. Do we need k8s min doc? We are dependent upon Pipelines LTS and follow whatever version it follows. Probably, we can add a new section later on.

Comment (savitaashture) [reply go.mod:3]: Yes I meant section for k8s version

**Review by savitaashture — APPROVED** [review]
/lgtm

**Pull request merged**